step-controller/step-controller-server/src/main/java/step/controller/grid/services/AbstractFunctionServices.java

```diff
@@ -127,6 +130,38 @@ public Function lookupCallFunction(CallFunction callFunction) {
 		return function;
 	}
 
+	@Operation(operationId = "lookupCall{Entity}WithBindings")
+	@POST
+	@Path("/lookup-with-bindings")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Secured(right="{entity}-read")
+	public Function lookupCallFunctionWithBindings(LookupCallFunctionRequest lookupCallFunctionRequest) {
+		Function function = null;
+		try {
+			ObjectPredicate objectPredicate = objectPredicateFactory.getObjectPredicate(getSession());
+			Session<User> session = getSession();
+			Map<String, Object> contextBindings = Objects.requireNonNullElse(lookupCallFunctionRequest.bindings, new HashMap<>());
+			if(!contextBindings.containsKey("user") && session!=null) {
```

Comment:
Do we want to allow changing the user i.e. performing a lookup for another user? for this service it doesn't seem to be very critical but in theory I would only allow admin users to do this